From b66796bcd888e0204955913e642d8e45937843dd Mon Sep 17 00:00:00 2001
From: Jan Cholasta <jcholast@redhat.com>
Date: Wed, 3 May 2017 06:12:36 +0000
Subject: [PATCH] certs: do not export CA certs in install_pem_from_p12

This fixes `kdc.crt` containing the full chain rather than just the KDC
certificate in CA-less server install.

https://pagure.io/freeipa/issue/6831
https://pagure.io/freeipa/issue/6869

Reviewed-By: Stanislav Laznicka <slaznick@redhat.com>
Reviewed-By: Martin Babinsky <mbabinsk@redhat.com>
---
 ipaserver/install/certs.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ipaserver/install/certs.py b/ipaserver/install/certs.py
index 06a7e2143964484fa45106ca381043eb440dc5b1..02c479d92511fcf4043e7d6798c85cf8256c3299 100644
--- a/ipaserver/install/certs.py
+++ b/ipaserver/install/certs.py
@@ -64,7 +64,7 @@ def get_cert_nickname(cert):
 
 def install_pem_from_p12(p12_fname, p12_passwd, pem_fname):
     pwd = ipautil.write_tmp_file(p12_passwd)
-    ipautil.run([paths.OPENSSL, "pkcs12", "-nokeys",
+    ipautil.run([paths.OPENSSL, "pkcs12", "-nokeys", "-clcerts",
                  "-in", p12_fname, "-out", pem_fname,
                  "-passin", "file:" + pwd.name])
 
-- 
2.9.4

